code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/VolumeDaoImpl.java

```diff
@@ -53,8 +53,12 @@ public Volume findVolumeByExternalId(Long storagePoolId, String externalId) {
     }
 
     @Override
-    public void createVolumeInStoragePool(Map<String, Object> volumeData, StoragePool storagePool) {
-        Volume volume = resourceDao.createAndSchedule(Volume.class, volumeData);
+    public void createVolumeInStoragePool(Map<String, Object> volumeData, String volumeName, StoragePool storagePool) {
```

Comment:
For other reviewers ref: all calls within this method will be executed as a part of same transaction as method is registered for this sort of execution here:

https://github.com/rancher/cattle/blob/4e25899064cd801a2066b856ed1e682d719a7579/code/iaas/model/src/main/resources/META-INF/cattle/system-services/spring-core-model-system-services-context.xml#L54